platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -217,7 +229,28 @@ void bufferOutput(RenderableOutputEvent output) {
                 lastUpdateTime = currentTimePeriod;
                 needHeaderSeparator = true;
             } else {
+                // We add the output to the buffer
+                // This won't consume significant memory as only the reference to `output` is stored
                 bufferedLogs.add(output);
+
+                // Update the approximate size of the buffered logs
+                if (output instanceof LogEvent) {
+                    LogEvent logEvent = (LogEvent) output;
+                    int logMessageCodepoints = logEvent.getMessage().length();
+                    approximateBufferCodepointSize += logMessageCodepoints;
+                } else if (output instanceof StyledTextOutputEvent) {
+                    StyledTextOutputEvent styledTextOutputEvent = (StyledTextOutputEvent) output;
+                    int logMessageCodepoints = styledTextOutputEvent
```

Comment:
🚀 How frequent is this operation? I'd avoid allocations here if it's very frequent

Reply:
This runs quite frequently: every grouped log line will hit this.
I didn't get a regression, but you have a point.

I think a classic for loop here would do the job just as well, and it would not cause an allocation.

WDYT?